### PR TITLE
Fix for mark as verified in Self Hosted

### PR DESCRIPTION
**Pull request by AlexTugarev**
fixes https://github.com/gitpod-io/gitpod/issues/2579

# how to test
1. do the initial setup
2. verify, there is no second tab opened once you select `Connect`.
  it should redirect to the login/terms flow first, and to the dashboard eventually
3. go to /access-control, you should be able to update permissions

### CODE REVIEW

File: components/server/src/auth/authenticator.ts [hunk 1]
```diff
@@ -124,15 +124,9 @@ export class Authenticator {
         }
 
         // prepare session
-        // hint: `returnToAfterTos` cannot be the referer, as we might be in a retry call.
-        const returnToAfterTos = this.env.hostUrl.withApi({
```
Comment (AlexTugarev) [inline components/server/src/auth/authenticator.ts:128]: a drive-by change. low-risk. remove an unnecessary indirection to /login.

File: components/server/src/env.ts [hunk 2]
```diff
@@ -89,8 +89,8 @@ export class Env extends AbstractComponentEnv {
     readonly authProviderConfigs = this.parseAuthProviderParamss();
 
     protected parseAuthProviderParamss(): AuthProviderParams[] {
-        const envVar = getEnvVar('AUTH_PROVIDERS_CONFIG');
-        return parseAuthProviderParamsFromEnv(JSON.parse(envVar));
+        // const envVar = getEnvVar('AUTH_PROVIDERS_CONFIG');
+        return [];
```
Comment (AlexTugarev) [inline components/server/src/env.ts:93]: to simulate a SH installation

Comment (geropl) [reply components/server/src/env.ts:93]: We should remove the content in the `values.yaml`.

File: components/server/src/auth/auth-provider-service.ts [hunk 3]
```diff
@@ -116,18 +116,25 @@ export class AuthProviderService {
         };
     }
 
-    async markAsVerified(params: { newOwnerId?: string; ownerId: string; id: string }) {
-        const { newOwnerId, ownerId, id } = params;
```
Comment (AlexTugarev) [inline components/server/src/auth/auth-provider-service.ts:120]: left over

File: components/server/src/env.ts (hunk 2 above)
Comment (AlexTugarev) [reply components/server/src/env.ts:93]: let's do that independently, ok?

Comment (geropl) [reply components/server/src/env.ts:93]: K, did already.

**Comment by geropl** [comment]
I was able to to the inital setup and login, but not to update permissions:
image

File: components/server/src/env.ts (hunk 2 above)
Comment (AlexTugarev) [reply components/server/src/env.ts:93]: @geropl what does this mean?

> K, did already.

**Comment by geropl** [comment]
After Clicking "Connect", I'm redirected back to /first-steps, including the half-filled auth provider form (client secret missing). On entering that again I'm correctly redirected to /tos and /access-control, though.

**Comment by AlexTugarev** [comment]
> @AlexTugarev After Clicking "Connect", I'm redirected back to /first-steps, including the half-filled auth provider form (client secret missing). On entering that again I'm correctly redirected to /tos and /access-control, though.

will check

File: components/server/src/auth/generic-auth-provider.ts [hunk 4]
```diff
@@ -293,64 +293,71 @@ export class GenericAuthProvider implements AuthProvider {
...
-                if (this.isOAuthError(err)) {
-                    message = 'OAuth Error. Please try again.'; // this is a 5xx response from authorization service
-                }
-                log.error(context, `(${strategyName}) Redirect to /sorry from verify callback`, err, { ...defaultLogPayload, err });
-                response.redirect(this.getSorryUrl(message));
-                return;
+        let result: Parameters<VerifyCallback>;
+        try {
+            result = await new Promise((resolve) => {
+                passport.authenticate(this.strategy as any, (...params: Parameters<VerifyCallback>) => resolve(params))(request, response, next);
```
Comment (svenefftinge) [inline components/server/src/auth/generic-auth-provider.ts:299]: should we really pass the `next` object here?

**Review by svenefftinge — APPROVED** [review]
